src/ParseObject.js

```diff
@@ -1700,8 +1700,23 @@ class ParseObject {
     if (!json.className) {
       throw new Error('Cannot create an object without a className');
     }
+
     const constructor = classMap[json.className];
-    const o = constructor ? new constructor() : new ParseObject(json.className);
+
+    // Remove reserved fields
+    const attributes = {}
+    Object.assign({}, json);
```

Comment:
if you wanted to clone `json`, just `const clonedJson = Object.assign({}, json)` and after refer to `clonedJson`. Otherwise you are mutating `json`